src/gluonts/mx/distribution/inflated_beta.py

```diff
@@ -0,0 +1,258 @@
+        "beta": 1,
+        "zero_probability": 1,
+        "one_probability": 1,
+    }
+    distr_cls: type = ZeroAndOneInflatedBeta
+
+    @classmethod
+    def domain_map(cls, F, alpha, beta, zero_probability, one_probability):
+        r"""
+        Maps raw tensors to valid arguments for constructing a ZeroAndOneInflatedBeta
+        distribution.
+
+        Parameters
+        ----------
+        F:
+        alpha:
+            Tensor of shape `(*batch_shape, 1)`
+        beta:
+            Tensor of shape `(*batch_shape, 1)`
+        zero_probability:
+            Tensor of shape `(*batch_shape, 1)`
+
+        Returns
+        -------
+        Tuple[Tensor, Tensor]:
+            Four squeezed tensors, of shape `(*batch_shape)`: First two have entries mapped to the
+            positive orthant, zero_probability is mapped to (0, 1), one_probability is mapped to (0, 1-zero_probability)
+        """
+        epsilon = np.finfo(cls._dtype).eps  # machine epsilon
+
+        alpha = softplus(F, alpha) + epsilon
+        beta = softplus(F, beta) + epsilon
+        zero_probability = F.sigmoid(zero_probability)
+        one_probability = (1 - zero_probability) * F.sigmoid(one_probability)
+        return (
+            alpha.squeeze(axis=-1),
+            beta.squeeze(axis=-1),
+            zero_probability.squeeze(axis=-1),
+            one_probability.squeeze(axis=-1),
+        )
+
+    @property
+    def event_shape(self) -> Tuple:
+        return ()
+
+    @property
+    def value_in_support(self) -> float:
+        return 0.5
+
+
+class ZeroInflatedBetaOutput(DistributionOutput):
+    args_dim: Dict[str, int] = {"alpha": 1, "beta": 1, "zero_probability": 1}
+    distr_cls: type = ZeroInflatedBeta
+
+    @classmethod
+    def domain_map(cls, F, alpha, beta, zero_probability):
+        r"""
+        Maps raw tensors to valid arguments for constructing a ZeroInflatedBeta
+        distribution.
+
+        Parameters
+        ----------
+        F:
+        alpha:
+            Tensor of shape `(*batch_shape, 1)`
+        beta:
+            Tensor of shape `(*batch_shape, 1)`
+        zero_probability:
+            Tensor of shape `(*batch_shape, 1)`
+
+        Returns
+        -------
+        Tuple[Tensor, Tensor]:
+            Three squeezed tensors, of shape `(*batch_shape)`: First two have entries mapped to the
+            positive orthant, last is mapped to (0,1)
+        """
+        epsilon = np.finfo(cls._dtype).eps  # machine epsilon
+
+        alpha = softplus(F, alpha) + epsilon
+        beta = softplus(F, beta) + epsilon
+        zero_probability = F.sigmoid(zero_probability)
+        return (
+            alpha.squeeze(axis=-1),
+            beta.squeeze(axis=-1),
+            zero_probability.squeeze(axis=-1),
+        )
+
+    @property
+    def event_shape(self) -> Tuple:
+        return ()
+
+    @property
+    def value_in_support(self) -> float:
+        return 0.5
+
+
+class OneInflatedBetaOutput(ZeroInflatedBetaOutput):
```

Comment:
Same as above: couldn't this inherit from `ZeroAndOneInflatedBetaOutput`?

Reply:
In the `OneInflatedBetaOutput` am using the domain map of the `ZeroInflatedBetaOutput`,
```
@classmethod
    def domain_map(cls, F, alpha, beta, one_probability):
        return super().domain_map(F, alpha, beta, one_probability)
```
so currently OIBO needs to inherit form ZIBO... Should I refactor that somehow?

Reply:
ah sorry, got it now